libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -37,6 +40,8 @@
 #include "AP_DDS_Service_Table.h"
 #include "AP_DDS_External_Odom.h"
 
+#define STRCPY(D,S) strncpy(D, S, sizeof(D))
```

Comment:
```suggestion
#define STRCPY(D,S) strncpy(D, S, ARRAY_SIZE(D))
```

You are doing the sizeof a pointer rather than the array size. Common mistake. Try this instead.